// permitio/build/openapi/types/authn-me-user-read.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/openapi/types/authn-me-user-read.ts
var authn_me_user_read_exports = {};
__export(authn_me_user_read_exports, {
  AuthnMeUserReadActorTypeEnum: () => AuthnMeUserReadActorTypeEnum,
  AuthnMeUserReadIsOnboardingEnum: () => AuthnMeUserReadIsOnboardingEnum,
  AuthnMeUserReadOnboardingStepEnum: () => AuthnMeUserReadOnboardingStepEnum
});
module.exports = __toCommonJS(authn_me_user_read_exports);
var AuthnMeUserReadActorTypeEnum = {
  User: "user"
};
var AuthnMeUserReadIsOnboardingEnum = {
  False: false
};
var AuthnMeUserReadOnboardingStepEnum = {
  Done: "done"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AuthnMeUserReadActorTypeEnum,
  AuthnMeUserReadIsOnboardingEnum,
  AuthnMeUserReadOnboardingStepEnum
});
