// permitio/build/openapi/types/elements-permission-level.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/openapi/types/elements-permission-level.ts
var elements_permission_level_exports = {};
__export(elements_permission_level_exports, {
  ElementsPermissionLevel: () => ElementsPermissionLevel
});
module.exports = __toCommonJS(elements_permission_level_exports);
var ElementsPermissionLevel = {
  Level1: "LEVEL_1",
  Level2: "LEVEL_2",
  Level3: "LEVEL_3",
  Level4: "LEVEL_4",
  Hidden: "HIDDEN",
  Unconfigured: "UNCONFIGURED"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ElementsPermissionLevel
});
