// permitio/build/openapi/types/onboarding-step.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/openapi/types/onboarding-step.ts
var onboarding_step_exports = {};
__export(onboarding_step_exports, {
  OnboardingStep: () => OnboardingStep
});
module.exports = __toCommonJS(onboarding_step_exports);
var OnboardingStep = {
  CreateOrganization: "create_organization",
  CreateProject: "create_project",
  CreateResource: "create_resource",
  CreateActions: "create_actions",
  AssignPermissions: "assign_permissions",
  AssignUserRoles: "assign_user_roles",
  ConnectSdk: "connect_sdk",
  Done: "done"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  OnboardingStep
});
