// permitio/build/openapi/types/programming-language.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/openapi/types/programming-language.ts
var programming_language_exports = {};
__export(programming_language_exports, {
  ProgrammingLanguage: () => ProgrammingLanguage
});
module.exports = __toCommonJS(programming_language_exports);
var ProgrammingLanguage = {
  Javascript: "javascript",
  Python: "python",
  Dotnet: "dotnet",
  Golang: "golang",
  Ruby: "ruby",
  Java: "java",
  KongGateway: "kong_gateway"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ProgrammingLanguage
});
