// permitio/build/openapi/api/audit-logs-api.mjs
// src/openapi/api/audit-logs-api.ts
import globalAxios2 from "axios";

// src/openapi/base.ts
import globalAxios from "axios";
var BASE_PATH = "http://localhost".replace(/\/+$/, "");
var BaseAPI = class {
  constructor(configuration, basePath = BASE_PATH, axios = globalAxios) {
    this.basePath = basePath;
    this.axios = axios;
    if (configuration) {
      this.configuration = configuration;
      this.basePath = configuration.basePath || this.basePath;
    }
  }
};
var RequiredError = class extends Error {
  constructor(field, msg) {
    super(msg);
    this.field = field;
    this.name = "RequiredError";
  }
};

// src/openapi/common.ts
var DUMMY_BASE_URL = "https://example.com";
var assertParamExists = function(functionName, paramName, paramValue) {
  if (paramValue === null || paramValue === void 0) {
    throw new RequiredError(
      paramName,
      `Required parameter ${paramName} was null or undefined when calling ${functionName}.`
    );
  }
};
var setBearerAuthToObject = async function(object, configuration) {
  if (configuration && configuration.accessToken) {
    const accessToken = typeof configuration.accessToken === "function" ? await configuration.accessToken() : await configuration.accessToken;
    object["Authorization"] = "Bearer " + accessToken;
  }
};
function setFlattenedQueryParams(urlSearchParams, parameter, key = "") {
  if (typeof parameter === "object") {
    if (Array.isArray(parameter)) {
      parameter.forEach((item) => setFlattenedQueryParams(urlSearchParams, item, key));
    } else {
      Object.keys(parameter).forEach(
        (currentKey) => setFlattenedQueryParams(
          urlSearchParams,
          parameter[currentKey],
          `${key}${key !== "" ? "." : ""}${currentKey}`
        )
      );
    }
  } else {
    if (urlSearchParams.has(key)) {
      urlSearchParams.append(key, parameter);
    } else {
      urlSearchParams.set(key, parameter);
    }
  }
}
var setSearchParams = function(url, ...objects) {
  const searchParams = new URLSearchParams(url.search);
  setFlattenedQueryParams(searchParams, objects);
  url.search = searchParams.toString();
};
var toPathString = function(url) {
  return url.pathname + url.search + url.hash;
};
var createRequestFunction = function(axiosArgs, globalAxios3, BASE_PATH2, configuration) {
  return (axios = globalAxios3, basePath = BASE_PATH2) => {
    const axiosRequestArgs = {
      ...axiosArgs.options,
      url: ((configuration == null ? void 0 : configuration.basePath) || basePath) + axiosArgs.url
    };
    return axios.request(axiosRequestArgs);
  };
};

// src/openapi/api/audit-logs-api.ts
var AuditLogsApiAxiosParamCreator = function(configuration) {
  return {
    /**
     *
     * @summary Get detailed audit log
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} logId The unique id of the audit log
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getDetailedAuditLog: async (projId, envId, logId, options = {}) => {
      assertParamExists("getDetailedAuditLog", "projId", projId);
      assertParamExists("getDetailedAuditLog", "envId", envId);
      assertParamExists("getDetailedAuditLog", "logId", logId);
      const localVarPath = `/v2/pdps/{proj_id}/{env_id}/audit_logs/{log_id}`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId))).replace(`{${"log_id"}}`, encodeURIComponent(String(logId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "GET", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    },
    /**
     *
     * @summary List Audit Logs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} [pdpId] Filter by pdp config id
     * @param {Array<string>} [users] List of user keys or emails to filter by
     * @param {boolean} [decision] Filter by decision result
     * @param {Array<string>} [resources] Filter by resources
     * @param {string} [tenant] Filter by tenant
     * @param {string} [action] Filter by action
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {AuditLogSortKey} [sortBy] Sort by column
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listAuditLogs: async (projId, envId, pdpId, users, decision, resources, tenant, action, timestampFrom, timestampTo, sortBy, page, perPage, options = {}) => {
      assertParamExists("listAuditLogs", "projId", projId);
      assertParamExists("listAuditLogs", "envId", envId);
      const localVarPath = `/v2/pdps/{proj_id}/{env_id}/audit_logs`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "GET", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      if (pdpId !== void 0) {
        localVarQueryParameter["pdp_id"] = pdpId;
      }
      if (users) {
        localVarQueryParameter["users"] = users;
      }
      if (decision !== void 0) {
        localVarQueryParameter["decision"] = decision;
      }
      if (resources) {
        localVarQueryParameter["resources"] = resources;
      }
      if (tenant !== void 0) {
        localVarQueryParameter["tenant"] = tenant;
      }
      if (action !== void 0) {
        localVarQueryParameter["action"] = action;
      }
      if (timestampFrom !== void 0) {
        localVarQueryParameter["timestamp_from"] = timestampFrom;
      }
      if (timestampTo !== void 0) {
        localVarQueryParameter["timestamp_to"] = timestampTo;
      }
      if (sortBy !== void 0) {
        localVarQueryParameter["sort_by"] = sortBy;
      }
      if (page !== void 0) {
        localVarQueryParameter["page"] = page;
      }
      if (perPage !== void 0) {
        localVarQueryParameter["per_page"] = perPage;
      }
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    }
  };
};
var AuditLogsApiFp = function(configuration) {
  const localVarAxiosParamCreator = AuditLogsApiAxiosParamCreator(configuration);
  return {
    /**
     *
     * @summary Get detailed audit log
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} logId The unique id of the audit log
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async getDetailedAuditLog(projId, envId, logId, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.getDetailedAuditLog(
        projId,
        envId,
        logId,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    },
    /**
     *
     * @summary List Audit Logs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} [pdpId] Filter by pdp config id
     * @param {Array<string>} [users] List of user keys or emails to filter by
     * @param {boolean} [decision] Filter by decision result
     * @param {Array<string>} [resources] Filter by resources
     * @param {string} [tenant] Filter by tenant
     * @param {string} [action] Filter by action
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {AuditLogSortKey} [sortBy] Sort by column
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async listAuditLogs(projId, envId, pdpId, users, decision, resources, tenant, action, timestampFrom, timestampTo, sortBy, page, perPage, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.listAuditLogs(
        projId,
        envId,
        pdpId,
        users,
        decision,
        resources,
        tenant,
        action,
        timestampFrom,
        timestampTo,
        sortBy,
        page,
        perPage,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    }
  };
};
var AuditLogsApiFactory = function(configuration, basePath, axios) {
  const localVarFp = AuditLogsApiFp(configuration);
  return {
    /**
     *
     * @summary Get detailed audit log
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} logId The unique id of the audit log
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getDetailedAuditLog(projId, envId, logId, options) {
      return localVarFp.getDetailedAuditLog(projId, envId, logId, options).then((request) => request(axios, basePath));
    },
    /**
     *
     * @summary List Audit Logs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} [pdpId] Filter by pdp config id
     * @param {Array<string>} [users] List of user keys or emails to filter by
     * @param {boolean} [decision] Filter by decision result
     * @param {Array<string>} [resources] Filter by resources
     * @param {string} [tenant] Filter by tenant
     * @param {string} [action] Filter by action
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {AuditLogSortKey} [sortBy] Sort by column
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listAuditLogs(projId, envId, pdpId, users, decision, resources, tenant, action, timestampFrom, timestampTo, sortBy, page, perPage, options) {
      return localVarFp.listAuditLogs(
        projId,
        envId,
        pdpId,
        users,
        decision,
        resources,
        tenant,
        action,
        timestampFrom,
        timestampTo,
        sortBy,
        page,
        perPage,
        options
      ).then((request) => request(axios, basePath));
    }
  };
};
var AuditLogsApi = class extends BaseAPI {
  /**
   *
   * @summary Get detailed audit log
   * @param {AuditLogsApiGetDetailedAuditLogRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AuditLogsApi
   */
  getDetailedAuditLog(requestParameters, options) {
    return AuditLogsApiFp(this.configuration).getDetailedAuditLog(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.logId,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
  /**
   *
   * @summary List Audit Logs
   * @param {AuditLogsApiListAuditLogsRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AuditLogsApi
   */
  listAuditLogs(requestParameters, options) {
    return AuditLogsApiFp(this.configuration).listAuditLogs(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.pdpId,
      requestParameters.users,
      requestParameters.decision,
      requestParameters.resources,
      requestParameters.tenant,
      requestParameters.action,
      requestParameters.timestampFrom,
      requestParameters.timestampTo,
      requestParameters.sortBy,
      requestParameters.page,
      requestParameters.perPage,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
};
export {
  AuditLogsApi,
  AuditLogsApiAxiosParamCreator,
  AuditLogsApiFactory,
  AuditLogsApiFp
};
