// permitio/build/openapi/api/decision-logs-api.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/openapi/api/decision-logs-api.ts
var decision_logs_api_exports = {};
__export(decision_logs_api_exports, {
  DecisionLogsApi: () => DecisionLogsApi,
  DecisionLogsApiAxiosParamCreator: () => DecisionLogsApiAxiosParamCreator,
  DecisionLogsApiFactory: () => DecisionLogsApiFactory,
  DecisionLogsApiFp: () => DecisionLogsApiFp
});
module.exports = __toCommonJS(decision_logs_api_exports);
var import_axios2 = __toESM(require("axios"));

// src/openapi/base.ts
var import_axios = __toESM(require("axios"));
var BASE_PATH = "http://localhost".replace(/\/+$/, "");
var BaseAPI = class {
  constructor(configuration, basePath = BASE_PATH, axios = import_axios.default) {
    this.basePath = basePath;
    this.axios = axios;
    if (configuration) {
      this.configuration = configuration;
      this.basePath = configuration.basePath || this.basePath;
    }
  }
};
var RequiredError = class extends Error {
  constructor(field, msg) {
    super(msg);
    this.field = field;
    this.name = "RequiredError";
  }
};

// src/openapi/common.ts
var DUMMY_BASE_URL = "https://example.com";
var assertParamExists = function(functionName, paramName, paramValue) {
  if (paramValue === null || paramValue === void 0) {
    throw new RequiredError(
      paramName,
      `Required parameter ${paramName} was null or undefined when calling ${functionName}.`
    );
  }
};
var setBearerAuthToObject = async function(object, configuration) {
  if (configuration && configuration.accessToken) {
    const accessToken = typeof configuration.accessToken === "function" ? await configuration.accessToken() : await configuration.accessToken;
    object["Authorization"] = "Bearer " + accessToken;
  }
};
function setFlattenedQueryParams(urlSearchParams, parameter, key = "") {
  if (typeof parameter === "object") {
    if (Array.isArray(parameter)) {
      parameter.forEach((item) => setFlattenedQueryParams(urlSearchParams, item, key));
    } else {
      Object.keys(parameter).forEach(
        (currentKey) => setFlattenedQueryParams(
          urlSearchParams,
          parameter[currentKey],
          `${key}${key !== "" ? "." : ""}${currentKey}`
        )
      );
    }
  } else {
    if (urlSearchParams.has(key)) {
      urlSearchParams.append(key, parameter);
    } else {
      urlSearchParams.set(key, parameter);
    }
  }
}
var setSearchParams = function(url, ...objects) {
  const searchParams = new URLSearchParams(url.search);
  setFlattenedQueryParams(searchParams, objects);
  url.search = searchParams.toString();
};
var toPathString = function(url) {
  return url.pathname + url.search + url.hash;
};
var createRequestFunction = function(axiosArgs, globalAxios3, BASE_PATH2, configuration) {
  return (axios = globalAxios3, basePath = BASE_PATH2) => {
    const axiosRequestArgs = {
      ...axiosArgs.options,
      url: ((configuration == null ? void 0 : configuration.basePath) || basePath) + axiosArgs.url
    };
    return axios.request(axiosRequestArgs);
  };
};

// src/openapi/api/decision-logs-api.ts
var DecisionLogsApiAxiosParamCreator = function(configuration) {
  return {
    /**
     *
     * @summary List PDP Decisions
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} pdpId The unique id of the pdp
     * @param {Array<string>} [users] List of user IDs to filter by
     * @param {boolean} [approved] Filter by approved decisions
     * @param {Array<string>} [resources] Filter by resources
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listPdpDecisionLogs: async (projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options = {}) => {
      assertParamExists("listPdpDecisionLogs", "projId", projId);
      assertParamExists("listPdpDecisionLogs", "envId", envId);
      assertParamExists("listPdpDecisionLogs", "pdpId", pdpId);
      const localVarPath = `/v2/pdps/{proj_id}/{env_id}/decision-logs/{pdp_id}`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId))).replace(`{${"pdp_id"}}`, encodeURIComponent(String(pdpId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "GET", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      if (users) {
        localVarQueryParameter["users"] = users;
      }
      if (approved !== void 0) {
        localVarQueryParameter["approved"] = approved;
      }
      if (resources) {
        localVarQueryParameter["resources"] = resources;
      }
      if (timestampFrom !== void 0) {
        localVarQueryParameter["timestamp_from"] = timestampFrom;
      }
      if (timestampTo !== void 0) {
        localVarQueryParameter["timestamp_to"] = timestampTo;
      }
      if (page !== void 0) {
        localVarQueryParameter["page"] = page;
      }
      if (perPage !== void 0) {
        localVarQueryParameter["per_page"] = perPage;
      }
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    }
  };
};
var DecisionLogsApiFp = function(configuration) {
  const localVarAxiosParamCreator = DecisionLogsApiAxiosParamCreator(configuration);
  return {
    /**
     *
     * @summary List PDP Decisions
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} pdpId The unique id of the pdp
     * @param {Array<string>} [users] List of user IDs to filter by
     * @param {boolean} [approved] Filter by approved decisions
     * @param {Array<string>} [resources] Filter by resources
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.listPdpDecisionLogs(
        projId,
        envId,
        pdpId,
        users,
        approved,
        resources,
        timestampFrom,
        timestampTo,
        page,
        perPage,
        options
      );
      return createRequestFunction(localVarAxiosArgs, import_axios2.default, BASE_PATH, configuration);
    }
  };
};
var DecisionLogsApiFactory = function(configuration, basePath, axios) {
  const localVarFp = DecisionLogsApiFp(configuration);
  return {
    /**
     *
     * @summary List PDP Decisions
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} pdpId The unique id of the pdp
     * @param {Array<string>} [users] List of user IDs to filter by
     * @param {boolean} [approved] Filter by approved decisions
     * @param {Array<string>} [resources] Filter by resources
     * @param {number} [timestampFrom] Filter by timestamp from
     * @param {number} [timestampTo] Filter by timestamp to
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options) {
      return localVarFp.listPdpDecisionLogs(
        projId,
        envId,
        pdpId,
        users,
        approved,
        resources,
        timestampFrom,
        timestampTo,
        page,
        perPage,
        options
      ).then((request) => request(axios, basePath));
    }
  };
};
var DecisionLogsApi = class extends BaseAPI {
  /**
   *
   * @summary List PDP Decisions
   * @param {DecisionLogsApiListPdpDecisionLogsRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof DecisionLogsApi
   */
  listPdpDecisionLogs(requestParameters, options) {
    return DecisionLogsApiFp(this.configuration).listPdpDecisionLogs(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.pdpId,
      requestParameters.users,
      requestParameters.approved,
      requestParameters.resources,
      requestParameters.timestampFrom,
      requestParameters.timestampTo,
      requestParameters.page,
      requestParameters.perPage,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DecisionLogsApi,
  DecisionLogsApiAxiosParamCreator,
  DecisionLogsApiFactory,
  DecisionLogsApiFp
});
