// permitio/build/openapi/api/proxy-config-api.mjs
// src/openapi/api/proxy-config-api.ts
import globalAxios2 from "axios";

// src/openapi/base.ts
import globalAxios from "axios";
var BASE_PATH = "http://localhost".replace(/\/+$/, "");
var BaseAPI = class {
  constructor(configuration, basePath = BASE_PATH, axios = globalAxios) {
    this.basePath = basePath;
    this.axios = axios;
    if (configuration) {
      this.configuration = configuration;
      this.basePath = configuration.basePath || this.basePath;
    }
  }
};
var RequiredError = class extends Error {
  constructor(field, msg) {
    super(msg);
    this.field = field;
    this.name = "RequiredError";
  }
};

// src/openapi/common.ts
var DUMMY_BASE_URL = "https://example.com";
var assertParamExists = function(functionName, paramName, paramValue) {
  if (paramValue === null || paramValue === void 0) {
    throw new RequiredError(
      paramName,
      `Required parameter ${paramName} was null or undefined when calling ${functionName}.`
    );
  }
};
var setBearerAuthToObject = async function(object, configuration) {
  if (configuration && configuration.accessToken) {
    const accessToken = typeof configuration.accessToken === "function" ? await configuration.accessToken() : await configuration.accessToken;
    object["Authorization"] = "Bearer " + accessToken;
  }
};
function setFlattenedQueryParams(urlSearchParams, parameter, key = "") {
  if (typeof parameter === "object") {
    if (Array.isArray(parameter)) {
      parameter.forEach((item) => setFlattenedQueryParams(urlSearchParams, item, key));
    } else {
      Object.keys(parameter).forEach(
        (currentKey) => setFlattenedQueryParams(
          urlSearchParams,
          parameter[currentKey],
          `${key}${key !== "" ? "." : ""}${currentKey}`
        )
      );
    }
  } else {
    if (urlSearchParams.has(key)) {
      urlSearchParams.append(key, parameter);
    } else {
      urlSearchParams.set(key, parameter);
    }
  }
}
var setSearchParams = function(url, ...objects) {
  const searchParams = new URLSearchParams(url.search);
  setFlattenedQueryParams(searchParams, objects);
  url.search = searchParams.toString();
};
var serializeDataIfNeeded = function(value, requestOptions, configuration) {
  const nonString = typeof value !== "string";
  const needsSerialization = nonString && configuration && configuration.isJsonMime ? configuration.isJsonMime(requestOptions.headers["Content-Type"]) : nonString;
  return needsSerialization ? JSON.stringify(value !== void 0 ? value : {}) : value || "";
};
var toPathString = function(url) {
  return url.pathname + url.search + url.hash;
};
var createRequestFunction = function(axiosArgs, globalAxios3, BASE_PATH2, configuration) {
  return (axios = globalAxios3, basePath = BASE_PATH2) => {
    const axiosRequestArgs = {
      ...axiosArgs.options,
      url: ((configuration == null ? void 0 : configuration.basePath) || basePath) + axiosArgs.url
    };
    return axios.request(axiosRequestArgs);
  };
};

// src/openapi/api/proxy-config-api.ts
var ProxyConfigApiAxiosParamCreator = function(configuration) {
  return {
    /**
     * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
     * @summary Create Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigCreate} proxyConfigCreate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createProxyConfig: async (projId, envId, proxyConfigCreate, options = {}) => {
      assertParamExists("createProxyConfig", "projId", projId);
      assertParamExists("createProxyConfig", "envId", envId);
      assertParamExists("createProxyConfig", "proxyConfigCreate", proxyConfigCreate);
      const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "POST", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      localVarHeaderParameter["Content-Type"] = "application/json";
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      localVarRequestOptions.data = serializeDataIfNeeded(
        proxyConfigCreate,
        localVarRequestOptions,
        configuration
      );
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    },
    /**
     * Deletes the proxy config and all its related data.
     * @summary Delete Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteProxyConfig: async (projId, envId, proxyConfigId, options = {}) => {
      assertParamExists("deleteProxyConfig", "projId", projId);
      assertParamExists("deleteProxyConfig", "envId", envId);
      assertParamExists("deleteProxyConfig", "proxyConfigId", proxyConfigId);
      const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId))).replace(`{${"proxy_config_id"}}`, encodeURIComponent(String(proxyConfigId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "DELETE", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    },
    /**
     * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
     * @summary Get Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProxyConfig: async (projId, envId, proxyConfigId, options = {}) => {
      assertParamExists("getProxyConfig", "projId", projId);
      assertParamExists("getProxyConfig", "envId", envId);
      assertParamExists("getProxyConfig", "proxyConfigId", proxyConfigId);
      const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId))).replace(`{${"proxy_config_id"}}`, encodeURIComponent(String(proxyConfigId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "GET", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    },
    /**
     * Lists all the proxy configs defined within an environment.
     * @summary List Proxy Configs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listProxyConfigs: async (projId, envId, page, perPage, options = {}) => {
      assertParamExists("listProxyConfigs", "projId", projId);
      assertParamExists("listProxyConfigs", "envId", envId);
      const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "GET", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      if (page !== void 0) {
        localVarQueryParameter["page"] = page;
      }
      if (perPage !== void 0) {
        localVarQueryParameter["per_page"] = perPage;
      }
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    },
    /**
     * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
     * @summary Update Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigUpdate} proxyConfigUpdate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateProxyConfig: async (projId, envId, proxyConfigId, proxyConfigUpdate, options = {}) => {
      assertParamExists("updateProxyConfig", "projId", projId);
      assertParamExists("updateProxyConfig", "envId", envId);
      assertParamExists("updateProxyConfig", "proxyConfigId", proxyConfigId);
      assertParamExists("updateProxyConfig", "proxyConfigUpdate", proxyConfigUpdate);
      const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`.replace(`{${"proj_id"}}`, encodeURIComponent(String(projId))).replace(`{${"env_id"}}`, encodeURIComponent(String(envId))).replace(`{${"proxy_config_id"}}`, encodeURIComponent(String(proxyConfigId)));
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }
      const localVarRequestOptions = { method: "PATCH", ...baseOptions, ...options };
      const localVarHeaderParameter = {};
      const localVarQueryParameter = {};
      await setBearerAuthToObject(localVarHeaderParameter, configuration);
      localVarHeaderParameter["Content-Type"] = "application/json";
      setSearchParams(localVarUrlObj, localVarQueryParameter);
      let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers
      };
      localVarRequestOptions.data = serializeDataIfNeeded(
        proxyConfigUpdate,
        localVarRequestOptions,
        configuration
      );
      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions
      };
    }
  };
};
var ProxyConfigApiFp = function(configuration) {
  const localVarAxiosParamCreator = ProxyConfigApiAxiosParamCreator(configuration);
  return {
    /**
     * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
     * @summary Create Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigCreate} proxyConfigCreate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async createProxyConfig(projId, envId, proxyConfigCreate, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.createProxyConfig(
        projId,
        envId,
        proxyConfigCreate,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    },
    /**
     * Deletes the proxy config and all its related data.
     * @summary Delete Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async deleteProxyConfig(projId, envId, proxyConfigId, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.deleteProxyConfig(
        projId,
        envId,
        proxyConfigId,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    },
    /**
     * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
     * @summary Get Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async getProxyConfig(projId, envId, proxyConfigId, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.getProxyConfig(
        projId,
        envId,
        proxyConfigId,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    },
    /**
     * Lists all the proxy configs defined within an environment.
     * @summary List Proxy Configs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async listProxyConfigs(projId, envId, page, perPage, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.listProxyConfigs(
        projId,
        envId,
        page,
        perPage,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    },
    /**
     * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
     * @summary Update Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigUpdate} proxyConfigUpdate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options) {
      const localVarAxiosArgs = await localVarAxiosParamCreator.updateProxyConfig(
        projId,
        envId,
        proxyConfigId,
        proxyConfigUpdate,
        options
      );
      return createRequestFunction(localVarAxiosArgs, globalAxios2, BASE_PATH, configuration);
    }
  };
};
var ProxyConfigApiFactory = function(configuration, basePath, axios) {
  const localVarFp = ProxyConfigApiFp(configuration);
  return {
    /**
     * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
     * @summary Create Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigCreate} proxyConfigCreate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createProxyConfig(projId, envId, proxyConfigCreate, options) {
      return localVarFp.createProxyConfig(projId, envId, proxyConfigCreate, options).then((request) => request(axios, basePath));
    },
    /**
     * Deletes the proxy config and all its related data.
     * @summary Delete Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteProxyConfig(projId, envId, proxyConfigId, options) {
      return localVarFp.deleteProxyConfig(projId, envId, proxyConfigId, options).then((request) => request(axios, basePath));
    },
    /**
     * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
     * @summary Get Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProxyConfig(projId, envId, proxyConfigId, options) {
      return localVarFp.getProxyConfig(projId, envId, proxyConfigId, options).then((request) => request(axios, basePath));
    },
    /**
     * Lists all the proxy configs defined within an environment.
     * @summary List Proxy Configs
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {number} [page] Page number of the results to fetch, starting at 1.
     * @param {number} [perPage] The number of results per page (max 100).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listProxyConfigs(projId, envId, page, perPage, options) {
      return localVarFp.listProxyConfigs(projId, envId, page, perPage, options).then((request) => request(axios, basePath));
    },
    /**
     * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
     * @summary Update Proxy Config
     * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
     * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
     * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
     * @param {ProxyConfigUpdate} proxyConfigUpdate
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options) {
      return localVarFp.updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options).then((request) => request(axios, basePath));
    }
  };
};
var ProxyConfigApi = class extends BaseAPI {
  /**
   * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
   * @summary Create Proxy Config
   * @param {ProxyConfigApiCreateProxyConfigRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof ProxyConfigApi
   */
  createProxyConfig(requestParameters, options) {
    return ProxyConfigApiFp(this.configuration).createProxyConfig(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.proxyConfigCreate,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
  /**
   * Deletes the proxy config and all its related data.
   * @summary Delete Proxy Config
   * @param {ProxyConfigApiDeleteProxyConfigRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof ProxyConfigApi
   */
  deleteProxyConfig(requestParameters, options) {
    return ProxyConfigApiFp(this.configuration).deleteProxyConfig(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.proxyConfigId,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
  /**
   * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
   * @summary Get Proxy Config
   * @param {ProxyConfigApiGetProxyConfigRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof ProxyConfigApi
   */
  getProxyConfig(requestParameters, options) {
    return ProxyConfigApiFp(this.configuration).getProxyConfig(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.proxyConfigId,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
  /**
   * Lists all the proxy configs defined within an environment.
   * @summary List Proxy Configs
   * @param {ProxyConfigApiListProxyConfigsRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof ProxyConfigApi
   */
  listProxyConfigs(requestParameters, options) {
    return ProxyConfigApiFp(this.configuration).listProxyConfigs(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.page,
      requestParameters.perPage,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
  /**
   * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
   * @summary Update Proxy Config
   * @param {ProxyConfigApiUpdateProxyConfigRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof ProxyConfigApi
   */
  updateProxyConfig(requestParameters, options) {
    return ProxyConfigApiFp(this.configuration).updateProxyConfig(
      requestParameters.projId,
      requestParameters.envId,
      requestParameters.proxyConfigId,
      requestParameters.proxyConfigUpdate,
      options
    ).then((request) => request(this.axios, this.basePath));
  }
};
export {
  ProxyConfigApi,
  ProxyConfigApiAxiosParamCreator,
  ProxyConfigApiFactory,
  ProxyConfigApiFp
};
